lib/queuePopulator/LogReader.js

```diff
@@ -379,6 +390,47 @@ class LogReader {
         return undefined;
     }
 
+    _processFilterEntry(batchState, record, done) {
+        const { entriesToPublish } = batchState;
+        if (!record || !record.entries) {
+            return done();
+        }
+        this._setEntryBatch(entriesToPublish);
+        return this._filterEntries(batchState, record, err => {
+            this._unsetEntryBatch();
+            if (err) {
+                this.log.error('error filtering entries from log', {
+                    method: 'LogReader._processFilterEntries',
+                    error: err,
+                });
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    _processFilterEntries(batchState, done) {
+        const { currentRecords } = batchState;
+        if (!currentRecords || !currentRecords.length > 0) {
```

Comment:
This contains a logic error due to priority of operators: `!currentRecords.length > 0` is equivalent to `(!currentRecords.length) > 0`. The end result may or not be affected in this particular case, but it should be fixed anyway.

Reply:
Ah I corrected this, didn't push it..will update.